File: app/src/main/java/com/example/doan/Models/LoginResponse.java
package com.example.doan.Models;

import com.google.gson.annotations.SerializedName;

public class LoginResponse {

    @SerializedName("userId")
    private int userId;

    @SerializedName("username")
    private String username;

    @SerializedName("fullName")
    private String fullName;

    @SerializedName("phone")
    private String phone;

    @SerializedName("role")
    private String role;

    @SerializedName("memberTier")
    private String memberTier;

    @SerializedName("token")
    private String token;

    public LoginResponse() {}

    public LoginResponse(int userId, String username, String fullName, String phone, String role, String memberTier, String token) {
        this.userId = userId;
        this.username = username;
        this.fullName = fullName;
        this.phone = phone;
        this.role = role;
        this.memberTier = memberTier;
        this.token = token;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getMemberTier() {
        return memberTier;
    }

    public void setMemberTier(String memberTier) {
        this.memberTier = memberTier;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public boolean isManager() {
        return "MANAGER".equals(role);
    }
}

File: app/src/main/java/com/example/doan/Models/LoginResponse.kt
package com.example.doan.Models

import com.google.gson.annotations.SerializedName

data class LoginResponse(
    @SerializedName("id")
    var userId: Int = 0,

    @SerializedName("username")
    var username: String? = null,

    @SerializedName("email")
    var email: String? = null,

    @SerializedName("fullName")
    var fullName: String? = null,

    @SerializedName("phone")
    var phone: String? = null,

    @SerializedName("role")
    var role: String? = null,

    @SerializedName("memberTier")
    var memberTier: String? = null,

    @SerializedName("accessToken") // Sửa lại để khớp với server
    var token: String? = null, // Đây là Access Token

    @SerializedName("refreshToken")
    var refreshToken: String? = null,

    @SerializedName("avatarUrl")
    var avatar: String? = null
) {
    fun isManager(): Boolean = role == "MANAGER"
}
